rwengine/src/audio/SoundEffect.hpp

```diff
@@ -0,0 +1,38 @@
+#ifndef SOUNDEFFECT_H
+#define SOUNDEFFECT_H
+
+#include <al.h>
+
+/**
+ * Class to represent any effect.
+ *
+ * Any concrete realisation should have own class
+ * in order to be able to set different effect-specific parameters.
+ */
+class SoundEffect {
+public:
+    /**
+     * Create effect
+     * @param type OpenAl specific effect type.
+     */
+    SoundEffect(ALint type);
+    ~SoundEffect();
```

Comment:
You should make the destructor virtual:
https://stackoverflow.com/questions/461203/when-to-use-virtual-destructors